### ISSUE TITLE
[Feature request] Auto backup

### ISSUE THREAD

**Issue by pawlosck** [TeamNewPipe/NewPipe#3558]
I would like to request for auto backup feature. Backup should be started before some actions, like adding new subscriptions, upgrading app (this is probably impossibly) or every day/week etc.

After last upgrade (19.3) newpipe crashed after start and I couldn't make backup before reinstall app (uninstall and next install), so I lost almost all subscriptions and whole history.

**Comment by Stypox** [comment]
See #3157

**Comment by KraXen72** [comment]
this would be a great feature! especially with the local-first nature of newpipe, there's too much crucial info stored on the device that could be lost with the device.
with this implemented, one can easily set up something like syncthing to always have the auto backups synced outside of the device.

i suggest being able to set the following:
- backup folder (through folder picker, so user can grant permission to sdcard folder as well)
- frequency of backups (12hr, daily, every 2 days, weekly, every 2 weeks)
- max amount of backups kept at a time (2-5) (the oldest ones get auto-deleted)

**Comment by swever826** [comment]
Yes I was looking for this feature! Also I would like to add that it would be helpful for us to be able to set the name of the backup file (so the old backup file can be replaced by the new one) or add a limit of how many backups we want in the folder.

Some of us are backing up the backup files to the cloud and the files can pile up quickly if they are set to be daily.

**Comment by delbonis** [comment]
I just fell victim to the Pixel 5a black screen of death and apparently hadn't made a backup in 8 months, so I'm a little sad rn.  Would have loved this.

**Comment by thiswillbeyourgithub** [comment]
I made a dumb adb script to regularly export newpipe if anyone cares: https://github.com/thiswillbeyourgithub/adb_newpipe_exporter
